### PR TITLE
add new article for windows and semgrep

### PR DESCRIPTION
**Pull request by sebasrevuelta**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [ ] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs
- [ ] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

**Comment by armchairlinguist** [comment]
@sebasrevuelta do we really need this article? 

The options are documented at https://semgrep.dev/docs/getting-started/, I think we could just add a small callout there saying something to the effect of "WSL and docker are your only options on windows". @s-santillan do you agree or would you rather not add more content on that page to keep it simple?

**Comment by sebasrevuelta** [comment]
Thanks for your feedback @armchairlinguist .
I wanted to do this article because there were some threads in the general channel asking for it, such as:
https://semgrep.slack.com/archives/CK86BJ5DW/p1689771137729439
https://semgrep.slack.com/archives/CK86BJ5DW/p1689942594695829?thread_ts=1689941037.070779&cid=CK86BJ5DW

and some PoC asking for Visual Studio Code on Windows, which I think is the good news of this article as I think people believe VSCode/Semgrep/Windows is not supported. 

In any case, I can change/adapt/forget the article in whatever direction.

**Comment by s-santillan** [comment]
Howdy! My primary development machine is actually Windows (I used to need DirectX for an old game engine in pre-Unity times and never moved to Mac.) I've been running Semgrep this whole time on a very custom WSL setup with its own X window system and a bunch of configs.

What I'd keep in the KB is **Running Semgrep for the VSCode extension**. The other sections I agree we can move to the getting started page.

There are gold nuggets there, especially the VSCode WSL extension in line 57. 

Just curious, @sebasrevuelta does the docker method also work with VSCode?

**Comment by sebastianrevuelta** [comment]
Hi team! I have simplified the KB. Now, it shows only the content for Visual Studio Code.  ;-)
@s-santillan about your question: Just curious, @sebasrevuelta does the docker method also work with VSCode? I tried it but with no success. ;-(

File: docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+description: Use this reference to run Visual Studio Code on Windows systems with the Semgrep extension.
+tags:
+  - Windows
+  - Visual Studio Code
+  - WSL
+---
+
+# Visual Studio Code and Semgrep on Windows
+Semgrep has developed an [extension](https://semgrep.dev/docs/extensions/semgrep-vs-code/) to be used in Visual Studio Code. 
+The good news is that it can be used in a Visual Studio Code installed on Windows.
```
Comment (armchairlinguist) [inline docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md:11]:
```suggestion
The extension can be used in Visual Studio Code installed on Windows, but you'll need to install WSL and add Semgrep to it.
```

File: docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+The steps to install and use it are as follow:
```
Comment (armchairlinguist) [inline docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md:12]:
```suggestion
Follow these steps to set up WSL, Semgrep, and the VS Code extension:
```

File: docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+6. Install Semgrep extension for Visual Studio Code. Check [how to install an extension.](https://code.visualstudio.com/docs/editor/extension-marketplace#_install-an-extension)
+7. Open the project from the Windows Command line with commands as `code /mnt/c/PATH_TO_TO_YOUT_PROJECT`
+
+You can scan with Semgrep now, clicking on the Semgrep icon in the right down corner!
+
+![image info](/img/kb/vscode-windows.png)
+
+> **Note:** To write this article a Windows 10 was used to test.
+
+> **Note:** As the date to write this article (August 2023) install Semgrep as a Python package is not supported, please, check in the [Semgrep documentation](https://semgrep.dev/docs/getting-started/#installing-and-running-semgrep-locally) if it is already supported.
```
Comment (armchairlinguist) [inline docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md:28]: I recommend removing this line. We should just update the docs and KB if this process is no longer required.

File: docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+4. Install Visual Studio Code on Windows, you can use [this guide.](https://code.visualstudio.com/docs/setup/windows)
+5. Install WSL extension for Visual Studio Code. Check [these steps.](https://code.visualstudio.com/docs/remote/wsl)
+6. Install Semgrep extension for Visual Studio Code. Check [how to install an extension.](https://code.visualstudio.com/docs/editor/extension-marketplace#_install-an-extension)
+7. Open the project from the Windows Command line with commands as `code /mnt/c/PATH_TO_TO_YOUT_PROJECT`
+
+You can scan with Semgrep now, clicking on the Semgrep icon in the right down corner!
+
+![image info](/img/kb/vscode-windows.png)
+
+> **Note:** To write this article a Windows 10 was used to test.
```
Comment (armchairlinguist) [inline docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md:26]:
```suggestion
:::info
This setup has been tested on Windows 10 with VS Code <X> and Semgrep <Y>. If you're having trouble with installation on other versions, [reach out to us](/docs/support). 
:::
```

Comment (armchairlinguist) [reply docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md:26]: X and Y would need replacement here before final release :)

File: docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+# Visual Studio Code and Semgrep on Windows
+Semgrep has developed an [extension](https://semgrep.dev/docs/extensions/semgrep-vs-code/) to be used in Visual Studio Code. 
+The good news is that it can be used in a Visual Studio Code installed on Windows.
+The steps to install and use it are as follow:
+
+1. Install WSL. 
+2. Install [Semgrep on WSL as Python package.](https://semgrep.dev/docs/getting-started/#installing-and-running-semgrep-locally)
+3. Run `semgrep login` to get the login URL for the Semgrep Cloud Platform. Open the login URL in the browser and login.
+4. Install Visual Studio Code on Windows, you can use [this guide.](https://code.visualstudio.com/docs/setup/windows)
```
[the remark this answers is not in the document]
Comment (sebasrevuelta) [reply docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md:17]: I will remove this line

File: docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md (hunk 3 above)
Comment (sebasrevuelta) [reply docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md:30]: I will remove this line

**Review by s-santillan — APPROVED** [review]
@sebasrevuelta LGTM! ~~Can I make a branch of this and add to "getting started with Semgrep CLI" some of your work in the 1st draft?~~ I tagged you as reviewer for this PR [https://github.com/returntocorp/semgrep-docs/pull/1160](https://github.com/returntocorp/semgrep-docs/pull/1160)

~~I can either make a separate PR or we can add to this one.~~

File: docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+5. Install WSL extension for Visual Studio Code. Check [these steps.](https://code.visualstudio.com/docs/remote/wsl)
+6. Install Semgrep extension for Visual Studio Code. Check [how to install an extension.](https://code.visualstudio.com/docs/editor/extension-marketplace#_install-an-extension)
+7. Open the project from the Windows Command line with commands as `code /mnt/c/PATH_TO_TO_YOUT_PROJECT`
```
Comment (s-santillan) [inline docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md]: @sebasrevuelta Just checking - are they always required to run VSCode from the Windows Command Line? They can't start VSCode from the start menu or a shortcut?

Comment (sebasrevuelta) [reply docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md]: Hey @s-santillan, they can start VSCode from the start menu or a shortcut. It is NOT mandatory to start VSCode from the command line.

Comment (armchairlinguist) [reply docs/kb/semgrep-cloud-platform/running-semgrep-on-windows.md]: Maybe we can just say "Open the project in VS Code" then?

**Pull request merged**